### DISCUSSION TITLE
Process archive by chunks for low memory usage

### DISCUSSION THREAD

**Discussion by ShellCode33** [hughsie/python-cabarchive#11] [Ideas]
Hey, thanks a lot for your work ! I'm working with huge CAB archives (few GBs), unfortunately your lib seems to load the entire thing in memory, which is not acceptable for my use case.

I don't have time to work on this right now but I thought I would still let you know. Maybe someone reading this will be willing to implement it

Cheers

**Comment by hughsie** [comment]
I guess we could rewrite things to chunk, but maybe a ready made thing would be to use the streaming interface of GCab? I've used it with gobject-introspection in the past, and although I've only every launched it from a blob or memory -- it should support loading from a stream too.
